pgx/src/datum/numeric_support/convert_anynumeric.rs

```diff
@@ -0,0 +1,121 @@
+//! Conversion implementations for from a thing into [AnyNumeric]
+use core::str::FromStr;
+use std::ffi::CStr;
+
+use pg_sys::AsPgCStr;
+
+use crate::numeric_support::call_numeric_func;
+use crate::numeric_support::convert::from_primitive_helper;
+use crate::numeric_support::error::Error;
+use crate::{pg_sys, AnyNumeric, IntoDatum, Numeric};
+
+impl<const P: u32, const S: u32> From<Numeric<P, S>> for AnyNumeric {
+    #[inline]
+    fn from(n: Numeric<P, S>) -> Self {
+        n.0
+    }
+}
+
+macro_rules! anynumeric_from_signed {
+    ($ty:ty, $as_:ty, $func:ident) => {
+        impl From<$ty> for AnyNumeric {
+            #[inline]
+            fn from(value: $ty) -> Self {
+                call_numeric_func(pg_sys::$func, vec![(value as $as_).into_datum()])
+            }
+        }
+    };
+}
+
+macro_rules! anynumeric_from_oversized_primitive {
+    ($ty:ty, $signed:ty) => {
+        impl From<$ty> for AnyNumeric {
+            #[inline]
+            fn from(value: $ty) -> Self {
+                match <$signed>::try_from(value) {
+                    Ok(value) => AnyNumeric::from(value),
+                    Err(_) => unsafe {
+                        // SAFETY:  conversion from value.to_string() will always work
+                        AnyNumeric::try_from(value.to_string().as_str()).unwrap_unchecked()
```

Comment:
Unsure the `unwrap_unchecked()` here is worth the risk, given that savings are quite small compered to the cost of string conversion. But it's your call.

Reply:
I suppose that's fair.